patches/minecraft/net/minecraft/fluid/FlowingFluid.java.patch

```diff
@@ -0,0 +1,11 @@
+--- a/net/minecraft/fluid/FlowingFluid.java
++++ b/net/minecraft/fluid/FlowingFluid.java
+@@ -159,7 +159,7 @@
+          BlockState blockstate = p_205576_1_.func_180495_p(blockpos);
+          FluidState fluidstate = blockstate.func_204520_s();
+          if (fluidstate.func_206886_c().func_207187_a(this) && this.func_212751_a(direction, p_205576_1_, p_205576_2_, p_205576_3_, blockpos, blockstate)) {
+-            if (fluidstate.func_206889_d()) {
++            if (fluidstate.func_206889_d() && net.minecraftforge.event.ForgeEventFactory.canCreateFluidSource((World) p_205576_1_, p_205576_2_, blockstate, this.func_205579_d())) {
```

Comment:
This is not the correct place to insert the patch. Move it below to where `if (this.canSourcesMultiply() && j >= 2)` is checked, and replace it with `j >= 2 && if (j >= 2 && net.minecraftforge.event.ForgeEventFactory.canCreateFluidSource(worldIn, pos, blockStateIn, canSourcesMultiply()))`, so you can also use the event the other way around (making fluid sources for blocks that normally don't make them).
It would also be better if we could just change the method to accept IWorldReader instead of casting to World.